sklearnex/covariance/incremental_covariance.py

```diff
@@ -117,14 +232,71 @@ def fit(self, X, queue=None):
         self : object
             Returns the instance itself.
         """
-        n_samples, n_features = X.shape
-        if self.batch_size is None:
-            batch_size_ = 5 * n_features
+
+        dispatch(
+            self,
+            "fit",
+            {
+                "onedal": self.__class__._onedal_fit,
+                "sklearn": None,
+            },
+            X,
+        )
+        return self
+
+    def _onedal_fit(self, X, queue=None):
+        self.n_samples_seen_ = 0
+
+        if sklearn_check_version("1.2"):
+            self._validate_params()
+
+        # finite check occurs on onedal side
+        if sklearn_check_version("1.0"):
+            X = self._validate_data(
+                X, dtype=[np.float64, np.float32], copy=self.copy, force_all_finite=False
+            )
         else:
-            batch_size_ = self.batch_size
-        for batch in gen_batches(n_samples, batch_size_):
+            X = check_array(
+                X, dtype=[np.float64, np.float32], copy=self.copy, force_all_finite=False
+            )
+            self.n_features_in_ = X.shape[1]
```

Comment:
```suggestion
        self.n_features_in_ = X.shape[1]
```

Reply:
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/base.py#L572 When reset=True for validate_data, it will set that parameter